src/sql/azdata.proposed.d.ts

```diff
@@ -921,4 +921,12 @@ declare module 'azdata' {
 		 */
 		physicalMemoryInMb?: number;
 	}
+
+	export interface NodeInfo {
+		/**
+		 * Path to the icon for the node. This property is optional.
+		 * Use this property if none of the icons defined in {@link SqlThemeIcon} fits the node.
+		 */
+		iconPath?: IconPath;
```

Comment:
I'm not sure I like having two separate properties for the icon. Could we at least make a general `icon` property that can take a string, SqlThemeIcon or IconPath and then when we promote that up to stable we can deprecate the current iconType property?

Reply:
good idea, I will make this property `icon` and add SqlThemeIcon as an optional type. currently iconType also has string as a possible type, but I don't like it, that means extensions needs to know the internal ADS class names. for the new property, the string type will be for the path of the icon.